metrics/web/lib/dashboard/presentation/state/project_metrics_notifier.dart

```diff
@@ -69,34 +64,27 @@ class ProjectMetricsNotifier extends ChangeNotifier {
   }
 
   /// Provides an error description that occurred during loading metrics data.
-  String get errorMessage => _errorMessage;
-
-  /// Subscribes to projects and its metrics.
-  Future<void> subscribeToProjects() async {
-    final projectsStream = _receiveProjectsUpdates();
-    _errorMessage = null;
-    await _projectsSubscription?.cancel();
-
-    _projectsSubscription = projectsStream.listen(
-      _projectsListener,
-      onError: _errorHandler,
-    );
-  }
-
-  /// Unsubscribes from projects and it's metrics.
-  Future<void> unsubscribeFromProjects() async {
-    await _cancelSubscriptions();
-    notifyListeners();
+  String get projectsErrorMessage => _projectsErrorMessage;
+
+  /// Subscribes to a projects name filter.
+  void subscribeToProjectsNameFilter() {
```

Comment:
Maybe we should subscribe to these updates in the constructor? And cancel it in the `dispose` method?